### ISSUE TITLE
CORS options

### ISSUE THREAD

**Issue by bumberboy** [bufbuild/connect-go#499]
Hello there!

I'm trying to build a go backend and a nodejs react frontend with connect. I have generated the server stubs for go and I use connect-query to generate the client.

However I'm having difficulty following the tutorial to enable cors on the go backend. Is there an option for this? I've tried creating an interceptor but I'm still getting `CORS missing allow origin` error.

```
func CorsInterceptor() connect.UnaryInterceptorFunc {
	interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
		return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
			req.Header().Set("Access-Control-Allow-Origin", "*")
			req.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS")
			req.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
			return next(ctx, req)
		}
	}
	return interceptor
}
```

Here is my buf.gen.yaml:
```
version: v1
plugins:
  - plugin: go
    out: gen/
    opt: paths=source_relative
  - plugin: connect-go
    out: gen/
    opt: paths=source_relative
  - plugin: buf.build/bufbuild/connect-query
    out: ./../frontend/gen
    opt: target=ts
  - plugin: buf.build/bufbuild/es
    out: ./../frontend/gen
    opt: target=ts
    ```

**Comment by qulaz** [comment]
@bumberboy use this to setup CORS https://github.com/bufbuild/connect-demo/blob/3a30d4de07d6ac42110acd4ebf64bb4bf8a62579/main.go#L88

**Comment by akshayjshah** [comment]
@qulaz has it right - CORS has to be handled at the `net/http` level, not in an interceptor. That's good, because you don't need anything connect-specific - you can use the same code/packages/etc that you'd use for CORS with any other HTTP handler. In the demo service, we use `github.com/rs/cors` and haven't had any problems with it.

Eventually, we intend to export some APIs to make this a bit easier. For now, the best approach is to copy the setup we use in our demo app.

**Comment by qulaz** [comment]
@akshayjshah I think it should be in the documentation because I spent a lot of time last time trying to figure it out.

First, like @bumberboy, I tried using an interceptor (similar to what I do with other frameworks). Then I tried reverse-engineering this Eliza demo on connect.build via the network tab in the browser.  Finally, I realized that this demo is available on GitHub and found how you manage CORS in it.
